hedera-node/hedera-app/src/itest/java/grpc/GrpcTransactionTest.java

```diff
@@ -0,0 +1,278 @@
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+package grpc;
+
+import static org.junit.jupiter.api.Assertions.*;
+
+import com.hedera.node.app.SessionContext;
+import com.hedera.node.app.grpc.GrpcServiceBuilder;
+import com.hedera.node.app.workflows.ingest.IngestWorkflow;
+import com.hedera.node.app.workflows.query.QueryWorkflow;
+import com.hederahashgraph.api.proto.java.*;
+import com.hederahashgraph.service.proto.java.ConsensusServiceGrpc;
+import com.hederahashgraph.service.proto.java.CryptoServiceGrpc;
+import com.hederahashgraph.service.proto.java.NetworkServiceGrpc;
+import io.grpc.ManagedChannelBuilder;
+import io.grpc.Status;
+import io.grpc.StatusRuntimeException;
+import io.helidon.grpc.client.ClientServiceDescriptor;
+import io.helidon.grpc.client.GrpcServiceClient;
+import io.helidon.grpc.server.GrpcRouting;
+import java.net.UnknownHostException;
+import java.nio.ByteBuffer;
+import javax.annotation.Nonnull;
+import org.junit.jupiter.api.BeforeEach;
+import org.junit.jupiter.api.DisplayName;
+import org.junit.jupiter.api.Test;
+
+/**
+ * This test generates and sends *real* GRPC transactions using the Google protobuf library and the
+ * Helidon gRPC client. This is basically a smoke-test. Nothing is done with these transactions,
+ * other than to make the call and verify that the call succeeds
+ */
+class GrpcTransactionTest extends GrpcTestBase {
+    private IngestWorkflow ingestWorkflow;
+    private QueryWorkflow queryWorkflow;
+
+    private GrpcServiceClient consensusClient;
+    private GrpcServiceClient cryptoClient;
+    private GrpcServiceClient networkClient;
+
+    @BeforeEach
+    @Override
+    void setUp() throws InterruptedException, UnknownHostException {
+        super.setUp();
+
+        final var channel = ManagedChannelBuilder.forAddress(host, port).usePlaintext().build();
+
+        consensusClient =
+                GrpcServiceClient.create(
+                        channel,
+                        ClientServiceDescriptor.builder(ConsensusServiceGrpc.getServiceDescriptor())
+                                .build());
+        networkClient =
+                GrpcServiceClient.create(
+                        channel,
+                        ClientServiceDescriptor.builder(NetworkServiceGrpc.getServiceDescriptor())
+                                .build());
+        cryptoClient =
+                GrpcServiceClient.create(
+                        channel,
+                        ClientServiceDescriptor.builder(CryptoServiceGrpc.getServiceDescriptor())
+                                .build());
+    }
+
+    @Override
+    protected void configureRouting(GrpcRouting.Builder rb) {
+        final var iw =
+                new IngestWorkflow() {
+                    @Override
+                    public void handleTransaction(
+                            @Nonnull SessionContext session,
```

Comment:
I ask myself if the annotation is good enough for such an inner implementation. I do not know if a tool like Spotbugs can really check this.